Assets/SEE/XR/KeyboardManager.cs

```diff
@@ -2,99 +2,112 @@
-/// This class manages the input for all buttons on the keyboard.
-/// This script is based on this tutorial: https://www.youtube.com/watch?v=vTonHBr4t4g
-/// </summary>
-public class KeyboardManager : MonoBehaviour
+namespace SEE.XR
 {
     /// <summary>
-    /// The instance of the keyboardmanager.
+    /// This class manages the input for all buttons on the keyboard.
+    /// This script is based on this tutorial: https://www.youtube.com/watch?v=vTonHBr4t4g
     /// </summary>
-    public static KeyboardManager instance;
-    /// <summary>
-    /// The shiftbutton, which performs the shift-action.
-    /// </summary>
-    public Button shiftButton;
-    /// <summary>
-    /// The deletebutton, which performs the delete-action.
-    /// </summary>
-    public Button deleteButton;
-    /// <summary>
-    /// The spacebuttom, which performs the space-action.
-    /// </summary>
-    public Button spaceButton;
-    /// <summary>
-    /// The enterbutton, which performs the enter-action.
-    /// </summary>
-    public Button enterButton;
-    /// <summary>
-    /// The inputfield, which is accessed by the keyboard.
-    /// </summary>
-    public TMP_InputField inputField;
-    /// <summary>
-    /// This image is used, to tell the user, that the shiftbutton is active.
-    /// </summary>
-    private Image shiftButtonImage;
-
-    /// <summary>
-    /// Is true, when the shiftbutton is active.
-    /// </summary>
-    private bool isShifted = false;
-    // Awake is always called before any Start functions.
-    private void Awake()
+    public class KeyboardManager : MonoBehaviour
     {
-        if (instance == null)
+        /// <summary>
+        /// The instance of the keyboardmanager.
+        /// </summary>
+        public static KeyboardManager instance;
+
+        /// <summary>
+        /// The shiftbutton, which shifts a character.
+        /// </summary>
+        public Button shiftButton;
+
+        /// <summary>
+        /// The deletebutton, which deletes a character.
+        /// </summary>
+        public Button deleteButton;
+
+        /// <summary>
+        /// The spacebuttom, which adds a space.
+        /// </summary>
+        public Button spaceButton;
+
+        /// <summary>
+        /// The enterbutton, which finalizes the input.
+        /// </summary>
+        public Button enterButton;
+
+        /// <summary>
+        /// The inputfield, which is accessed by the keyboard.
+        /// </summary>
+        public TMP_InputField inputField;
+
+        /// <summary>
+        /// This image is used to tell the user that the shiftbutton is active.
+        /// </summary>
+        private Image shiftButtonImage;
+
+        /// <summary>
+        /// Is true when the shiftbutton is active.
+        /// </summary>
+        private bool isShifted = false;
+
+        private void Awake()
         {
-            instance = this;
+            if (instance == null)
+            {
+                instance = this;
+            }
```

Comment:
This script is on top of the Tablet, which is always there in XR and will not be spawned multiple times or destroyed. So the code is just run once since it is the Awake() method.

Reply:
Yes, but it usually doesn't hurt to code defensively to guard against any future mistakes (either in setting up the scene, or in some other MonoBehaviour). We also do this in other singleton classes in SEE, for example, `WindowSpaceManager` or `IDEIntegration`.

Reply:
Alright, then I will move the code below the if condition in the if condition and it should be fine.

Reply:
I think I'd prefer it if we invert the if condition and either log an error and return, or throw an exception. Otherwise, it will fail silently and not leave a visible error if we do make such a mistake in the future.

Reply:
Please check if its alright now.

Reply:
I think we can replace the check with `instance != null` instead of searching the scene (since the object may not yet be present since we're only in `Awake` instead of `Start` like in the other singletons), but apart from that this looks good now 👍🏻